src/sentry/notifications/helpers.py

```diff
@@ -941,7 +869,87 @@ def get_layered_setting_providers(
     return user_to_setting
 
 
-def get_setting_providers_for_projects(
+###
+def get_setting_option_values(
+    recipients: Iterable[RpcActor | Team | User],
+    projects: Iterable[Project] | None = None,
+    organization: Organization | None = None,
+    additional_filters: Q | None = None,
+) -> MutableMapping[
+    RpcActor,
+    MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum],
+]:
+    user_to_setting = get_layered_setting_options(
```

Comment:
I'll add/improve docstrings in a followup PR with info about when to use the various functions below.